src/dual.jl

```diff
@@ -392,7 +392,7 @@ Base.float(d::Dual) = convert(float(typeof(d)), d)
 # General Mathematical Operations #
 ###################################
 
-for (M, f, arity) in DiffRules.diffrules()
+for (M, f, arity) in DiffRules.diffrules(filter_modules = (:Base, :SpecialFunctions, :NaNMath))
```

Comment:
I guess one could use the opportunity and already prepare this code for future releases of DiffRules. As stated in the deprecation warning it will change to `filter_modules=nothing`, ie. it will load all rules defined in DiffRules. We only had to introduce this keyword argument because many downstream packages, including ForwardDiff, did not actually check if a module `M` with a function `f` was defined when they tried to load the rules.

So an alternative approach here that would also add derivative definitions for functions in LogExpFunctions would be to:
- add LogExpFunctions >= 0.3 as explicit dependency (it's already an indirect dependency via DiffRules and SpecialFunctions)
- use `DiffRules.diffrules(; filter_modules=nothing)` and load all rules, including the ones for LogExpFunctions
- check if `M` and `M.f` are defined before trying to `eval` the rule definitions